powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/controller/api/PowerAuthControllerTest.java

```diff
@@ -125,4 +459,740 @@ void testUpdateActivation_badRequest() throws Exception {
+        final VerifyOfflineSignatureRequest verifyOfflineSignatureRequest =
+                new VerifyOfflineSignatureRequest();
+        verifyOfflineSignatureRequest.setActivationId(ACTIVATION_ID);
+        verifyOfflineSignatureRequest.setData(DATA);
+        verifyOfflineSignatureRequest.setSignature("123456");
+        verifyOfflineSignatureRequest.setAllowBiometry(false);
+
+        mockMvc.perform(post("/rest/v3/signature/offline/verify")
+                        .content(wrapInRequestObjectJson(verifyOfflineSignatureRequest))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.responseObject.signatureValid").value(false))
+                .andExpect(jsonPath("$.responseObject.activationId").value(ACTIVATION_ID));
+    }
+
+    /**
+     * Tests the retrieval of system status.
+     * <p>
+     * This test checks if the system status can be successfully retrieved without any errors.
+     *
+     * @throws Exception if the mockMvc.perform operation fails.
+     */
+    @Test
+    void testSystemStatus() throws Exception {
+        mockMvc.perform(post("/rest/v3/status")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    /**
+     * Tests the retrieval of a list of error codes.
+     * <p>
+     * This test requests a list of error codes and verifies that the list is populated with a sufficient number of entries.
+     *
+     * @throws Exception if the mockMvc.perform operation fails.
+     */
+    @Test
+    void testErrorList() throws Exception {
+        final GetErrorCodeListRequest getErrorCodeListRequest = new GetErrorCodeListRequest();
+        getErrorCodeListRequest.setLanguage(Locale.ENGLISH.getLanguage());
+        mockMvc.perform(post("/rest/v3/error/list")
+                        .content(wrapInRequestObjectJson(getErrorCodeListRequest))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.responseObject.errors.length()", greaterThan(32)));
+    }
+
+    /**
+     * Creates a request object for creating an operation.
+     * <p>
+     * This helper method constructs and returns an {@link OperationCreateRequest} with
+     * predefined application ID, template name, user ID, and proximity OTP settings.
+     *
+     * @param proximityOtp a boolean indicating whether proximity OTP is enabled
+     * @return a configured {@link OperationCreateRequest} instance
+     * @throws Exception if the operation creation request setup fails
+     */
+    private OperationCreateRequest createOperationCreateRequest(final boolean proximityOtp) throws Exception {
```

Comment:
Can by `static`? Moreover, let's rename the parameter, `proximityOtp` looks like OTP value to me.
```suggestion
    private static OperationCreateRequest createOperationCreateRequest(final boolean proximityOtpEnabled) throws Exception {
```